### ISSUE TITLE
Win10 server, Linux client - mouse not going over to other screen, gets lost

### ISSUE THREAD

**Issue by johnny-mac** [debauchee/barrier#1444]
### What happened?

Hey guys

So I encountered this problem when I upgraded to 2.4, I'm hanging back at 2.3.3 until this is resolved, every time I move my mouse from my Windows server to my Linux client, it disappears completely & I can't get it to come back. If I right-click while my mouse is in nowhereland it'll right click on the Linux server.  I have to ctrl + alt + delete to get my mouse back.

### Version

v2.4.0

### If applicable, where did you install Barrier from?

On windows from your .exe
On Linux from package manager

### What OSes are you seeing the problem on? (Check all that apply)

Linux, Windows

### What OS versions are you using?

Windows 10 pro & Artix Linux(rolling release)

**Comment by ShineSmile** [comment]
Same issue for 2.4.0 in Windows as server. Client side OS is OSX.

Mouse get stuck in the right bottom corner of  client desktop.
2.3.3 works well.
2.3.4 have some issue to download.

**Comment by Suppi123** [comment]
I have a similar problem. Windows 10 Server. Client side is Ubuntu 21.10

I do not see the mouse on the client side. But I can move it back to the server side. 
When I move the invisible mouse pointer over elements of the client side, they get a visible focus.

**Comment by sepehr78** [comment]
Same happens for Windows 10 as server and Ubuntu 20.04.3 LTS as client. Both running Release 2.4.0. When server is downgraded to 2.3.4, it works. So it appears that the issue stems from 2.4.0 on Windows 10.

**Comment by tbossi** [comment]
Same issue here for windows 11 as server (release 2.4.0), and Manjaro as client (release 2.4.0). When server is downgraded to 2.3.4 it works.

The issue seems limited to the mouse, because while the mouse should be on the client (but is lost), the keyboard works correctly on the client.

**Comment by nbowwow** [comment]
I think my issue is somewhat relative to yours

I have a Manjaro KDE (Server) trying to connect to ubuntu KDE (Client) It is saying on the logs that they are connected, and it does say mouse left screen and mouse entered the screen, it even leaves the border but that's quite it, My ubuntu gets nothing, nada, BUT if I were to switch to macOS it works perfectly fine.

**Comment by Esirn** [comment]
I switched the "scaling and layout" of Windows 10 to "100%" and it works. But I prefer to downgrade the server version to 2.3.3.

**Comment by mr-pooglyfoop** [comment]
Having the same issue when using Windows 10 pro as server and Raspbian 10 Buster as the client. Downgrading server to 2.3.3 works flawlessly.

**Comment by woday-dev** [comment]
same issue here, windows as server, centos 7 as client(2.4), cursor disappear when goto linux, downgrade windows version to 2.3.4 fixed.

**Comment by letsgo0** [comment]
> same issue here, windows as server, centos 7 as client(2.4), cursor disappear when goto linux, downgrade windows version to 2.3.4 fixed.

that's right. when I enter screen from win10 to ubuntu,  the cursor disappears. and that downgrading windows' version to 2.3.4 works.

**Comment by johnny-mac** [comment]
What can we do to help you, devs? Let us know

**Comment by Fatstrid** [comment]
I also ended up using version 2.3.4 on the server and client devices and the mouse issue resolved itself.

PC/server: Windows 10
Laptop/client: Pop-OS

**Comment by jondawes** [comment]
I've just run into this on a fresh install of v2.4.0, uninstalled and tried 2.3.4 based on this thread and it works fine.

Server: Win 10 Pro - 10.0.19043
Client: Win 10 Pro 0 10.0.19043
Installed from exe found here on the releases page.

Client was directly to the left of the server and the control would switch and appear in both server and client log but I'd lose the mouse. Right-click would work and happen at the bottom right screen, bringing up a context menu. Keyboard behaved as expected with alt-tab, tab and windows key enabling navigation. If I used the mouse on the client it would appear and move. Right-click from the server would work but I still couldn't move it. I didn't think to try a left-click on a button at the time.

Stopping Client would return control to the server machine. I didn't need to completely close the client instance.

Removed and did a clean 2.3.4 install on both and it worked properly first time.

**Comment by OmarShawky1** [comment]
> Same issue for 2.4.0 in Windows as server. Client side is OSX.
> 
> Mouse get stuck in the right bottom corner of client desktop. 2.3.3 works well for windows server side. 2.3.4 have some issue to download.

This is the real solution, After being stuck for quite a lot of time, I never thought that this can be the problem. However, making the scaling to 100% solved it entirely. Thank you sir.

**Comment by brdacost** [comment]
I'm trying to use a macOS 12.4 as Server and an Arch Linux x86_64 5.18.12-arch1-1 as client. The logs on my mac show:

```
started server (IPv4/IPv6), waiting for clients
2022-07-21 14:57:44.365107-0300 barriers[82747:4431184] starting cocoa loop
```

And then, when I connect to it with my linux, it shows a success message (that it has in fact recognized and connected to the client), but then, when I get to the edge of the mac's screen, it shows:

```
[2022-07-21T15:05:34] NOTE: accepted client connection
[2022-07-21T15:05:34] NOTE: client "archlinux" has connected
[2022-07-21T15:05:41] INFO: switch from "MacBook-Pro-de-Bruno.local" to "archlinux" at 0,496
[2022-07-21T15:05:41] INFO: leaving screen
[2022-07-21T15:05:41] WARNING: cursor may not be visible
```

Fresh installing it did not work (also the auto config did not work as well, I had to manually input the server's IP and disable the SSL on both client and server's settings).

Any help would be appreciated, thanks in advance :)

P.S.: using xorg instead of wayland solved the problem

**Comment by karlk79** [comment]
I'm having this problem too, will it be fixed in the next release?

**Comment by Thrakorzog** [comment]
In Ubuntu, on the login page, After you select your user account, but before you put in your password, there is a gear below on the right-hand side.  Selecting this gear allows you to choose the Display Engine, I have used multiple that work with Barrier, and the default (I assume Waylan) that doesn't fully work (it exhibits the same behavior of registering coming in and out, but the mouse isn't controllable).
